idea/src/org/jetbrains/kotlin/idea/intentions/ReplaceWithOrdinaryAssignmentIntention.kt

```diff
@@ -8,19 +8,28 @@ package org.jetbrains.kotlin.idea.intentions
 import com.intellij.codeInsight.intention.LowPriorityAction
 import com.intellij.openapi.editor.Editor
 import org.jetbrains.kotlin.idea.KotlinBundle
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.project.builtIns
 import org.jetbrains.kotlin.lexer.KtTokens
 import org.jetbrains.kotlin.psi.KtBinaryExpression
 import org.jetbrains.kotlin.psi.KtNameReferenceExpression
 import org.jetbrains.kotlin.psi.KtPsiFactory
 import org.jetbrains.kotlin.psi.createExpressionByPattern
+import org.jetbrains.kotlin.resolve.calls.callUtil.getType
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
 
 class ReplaceWithOrdinaryAssignmentIntention :
     SelfTargetingIntention<KtBinaryExpression>(KtBinaryExpression::class.java, KotlinBundle.message("replace.with.ordinary.assignment")),
     LowPriorityAction {
     override fun isApplicableTo(element: KtBinaryExpression, caretOffset: Int): Boolean {
         if (element.operationToken !in KtTokens.AUGMENTED_ASSIGNMENTS) return false
-        if (element.left !is KtNameReferenceExpression) return false
```

Comment:
Code which was there before and what you're writing now are both absolutely wrong. I changed issue title: https://youtrack.jetbrains.com/issue/KT-34715 and wrote comment in issue.

Right implementation does this:
1. Resolve augmented assignment
2. Check that it resolves to `plusAssign`, `minusAssign`, etc
3. If so then intention shouldn't be suggested

E.g. this intention should work for such cases (but it doesn't work for such cases even now):
```
class Foo {
    var x: Int = 1
        get() {
            println("getter")
            return field
        }
        set(value) {
            println("setter")
            field = value
        }
}

fun main() {
    val foo = Foo()
    foo.x += 1
}
```

Reply:
So will you continue working on this PR? I can fix this issue